crates/mirror-mirror/src/map.rs

```diff
@@ -4,6 +4,13 @@ use core::fmt;
 use crate::iter::PairIterMut;
 use crate::Reflect;
 
+/// A reflected map type.
+///
+/// Note this is only implemented for [`BTreeMap`] and _not_ [`HashMap`] due to technical
+/// limitations.
```

Comment:
Yes I'll expand it!

Its because we need a variant for map values. At first I wanted that to be `enum Value { Map(HashMap<Value, Value>) }`. However now `Value` is used as the key in a `HashMap` so it most implement `Hash + Eq`. However it can't since it contains a `HashMap` which doesn't implement `Hash + Eq`, because there is no stable iteration order 😞

Using `Map(BTreeMap<Value, Value>)` does work since we can make `Value` implement `Eq + Ord` by mapping it to a type that does https://github.com/EmbarkStudios/mirror-mirror/blob/main/crates/mirror-mirror/src/value.rs#L65-L90. We cannot just have `#[derive(Ord, Eq)]` on `Value` because it contains floats.

One possible solution is to store all maps as `BTreeMap<Value, Value>` regardless if the original value was a `HashMap`. However that feels kinda dirty so I didn't pursue that.